src/Examples/PiTop.Interactive.Rover/RoverRobotSetup.cs

```diff
@@ -0,0 +1,116 @@
+
+            var foundationExtension = new MakerArchitecture.Foundation.InteractiveExtension.KernelExtension();
+            await foundationExtension.OnLoadAsync(csharpKernel);
+
+            var expansionExtension = new MakerArchitecture.Expansion.InteractiveExtension.KernelExtension();
+            await expansionExtension.OnLoadAsync(csharpKernel);
+
+            PiTop4Board.Instance.UseCamera();
+            var rover = new RoverRobot(PiTop4Board.Instance.GetOrCreateExpansionPlate(), PiTop4Board.Instance.GetOrCreateCamera<StreamingCamera>(0),
+                RoverRobotConfiguration.Default);
+            var strategies = new RoverRobotStrategies();
+
+            rover.AllLightsOn();
+            rover.BlinkAllLights();
+
+            await RegisterTypes<Image>(csharpKernel);
+            await RegisterTypes<Color>(csharpKernel);
+            await RegisterTypes<Speed>(csharpKernel);
+            await RegisterTypes<PiTop4Board>(csharpKernel);
+            await RegisterTypes<StreamingCamera>(csharpKernel);
+            await RegisterTypes<FoundationPlate>(csharpKernel);
+            await RegisterTypes<ExpansionPlate>(csharpKernel);
+            await RegisterTypes<RoverRobot>(csharpKernel);
+            await csharpKernel.SetVariableAsync("roverRobot", rover);
+            await csharpKernel.SetVariableAsync("roverRobotStrategies", strategies);
+
+
+            var source = new CancellationTokenSource();
+
+            var robotLoop = Task.Run(() =>
+            {
+                while (!source.IsCancellationRequested)
+                {
+                    if (!source.IsCancellationRequested)
+                    {
+                        strategies.Perceive?.Invoke(rover, DateTime.Now, source.Token);
+                    }
+
+                    if (!source.IsCancellationRequested)
+                    {
+                        strategies.Plan?.Invoke(rover, DateTime.Now, source.Token);
+                    }
+
+                    if (!source.IsCancellationRequested)
+                    {
+                        strategies.Act?.Invoke(rover, DateTime.Now, source.Token);
+                    }
+                }
+
+                rover.MotionComponent.Stop();
+            }, source.Token);
+
+            var reactLoop = Task.Run(() =>
+            {
+                while (!source.IsCancellationRequested)
+                {
+                    if (!source.IsCancellationRequested)
+                    {
+                        strategies.React?.Invoke(rover, DateTime.Now, source.Token);
+                    }
+                }
+
+                rover.MotionComponent.Stop();
+            }, source.Token);
+
+            csharpKernel.RegisterForDisposal(() =>
+            {
+                source.Cancel(false);
+                Task.WaitAll(new []{robotLoop, reactLoop},TimeSpan.FromSeconds(10));
+                if (csharpKernel.TryGetVariable("roverRobot", out RoverRobot roverRobot))
```

Comment:
You have a reference to the instance above, so you shouldn't need to pull it out this way to dispose it.

Likely it could be registered up front for disposal as well.